docs/modules/vector-search/pages/vector-collections.adoc

```diff
@@ -0,0 +1,498 @@
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+VectorDocument<String> document = VectorDocument.of("some document value", VectorValues.of(new float[] {0.2f, 0.9f, -1.2f}));
+CompletionStage<Void> result = vectorCollection.setAsync("1", document);
+----
+--
+====
+
+=== Delete objects
+To delete document from a vector collection, use the method `deleteAsync` or `removeAsync` of the `VectorCollection` class.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+CompletionStage<Void> resultDelete = vectorCollection.deleteAsync("1");
+CompletionStage<VectorDocument<String>> resultRemove = vectorCollection.removeAsync("2");
+----
+--
+====
+
+NOTE: These methods do not delete index nodes but mark them as deleted. This may impact search speed and memory usage. To permanently remove nodes from the index, you must run index optimization after deletion. See <<optimize-collection, optimize method>>
+
+== Similarity search
+
+Vector search returns objects with vectors that are most similar to that of the query, based on specified metrics.
+When submitting a search query, you have the option to include one vector values representing your query embeddings and specify a limit on the number of results to retrieve.
+
+=== Single-index vector collection search
+
+For similarity search, use the `searchAsync` method of the `VectorCollection`.
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of(new float[] {0, 0}),
+        SearchOptions.builder()
+            .limit(10)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+=== Multi-index vector collection search
+
+For similarity search, use the `searchAsync` method of the `VectorCollection`, specifying the index name to search.
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+CompletionStage<SearchResults<String, String>> results = vectorCollection.searchAsync(
+        VectorValues.of("index_name", new float[] {0, 0}),
+        SearchOptions.builder()
+            .limit(10)
+            .includeVectors()
+            .includeValue()
+            .build()
+);
+----
+--
+====
+
+=== Similarity search options
+Search parameters are passed as an argument `searchOptions` to the searchAsync method.
+.Search options
+[cols="1,2,3a",options="header"]
+|===
+|Option|Description|Example
+
+|limit
+|How many vectors should be returned in search result
+|
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+var options = SearchOptions.builder().limit(10).build();
+----
+--
+====
+
+|includeValue
+|Include the user value in the search result.
+|
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+var options = SearchOptions.builder().includeValue().build();
```

Comment:
these partial examples are misleading and rely on default limit = 1. Do we really want to provide such trivial code? Naming method + 1 example with everything should be sufficient